versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CaffeineCacheBackend.java

```diff
@@ -145,6 +179,17 @@ public void put(@Nonnull String repositoryId, @Nonnull Obj obj) {
     putLocal(repositoryId, obj);
   }
 
+  @VisibleForTesting
+  void cachePut(CacheKeyValue key, CacheKeyValue value) {
+    var w = weigher(key, value);
+    if (currentWeight.get() + w < admitWeight) {
+      currentWeight.addAndGet(w);
+      cache.put(key, value);
+    } else {
+      rejections.incrementAndGet();
```

Comment:
How about also tracking the weight of rejected objects? Cf. `CaffeineStatsCounter.recordEviction(...)`

Reply:
Hm - not sure that's really interesting though.

Reply:
Caffeine's eviction metrics trace the weight of evicted objects. It's be nice to trace the weight of rejected objects here for symmetry (using the same `DistributionSummary` meter).

It is certainly an interesting metrics because the weight of entries values a lot in runtime.